internal/services/user/user/users_data_source.go

```diff
@@ -0,0 +1,312 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/folio-sec/terraform-provider-zoom/internal/provider/shared"
+	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/samber/lo"
+)
+
+var (
+	_ datasource.DataSource              = &tfDataSource{}
+	_ datasource.DataSourceWithConfigure = &tfDataSource{}
+)
+
+func NewUsersDataSource() datasource.DataSource {
+	return &tfDataSource{}
+}
+
+type tfDataSource struct {
+	crud *crud
+}
+
+func (d *tfDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	data, ok := req.ProviderData.(*shared.ProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected ProviderData Source Configure Type",
+			fmt.Sprintf("Expected *provider.ProviderData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	d.crud = newCrud(data.UserClient)
+}
+
+func (d *tfDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user_users"
+}
+
+func (d *tfDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	markdownSeparatorForList := "\n  "
+
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `Gets basic information for multiple Zoom users.
+
+## API Permissions
+
+The following API permissions are required in order to use this resource.
+This resource requires the ` + strings.Join([]string{
+			"`user:read:list_users:admin`",
+		}, ", ") + ".",
+		Attributes: map[string]schema.Attribute{
+			"status": schema.StringAttribute{
```

Comment:
how about to add `filter` SingleNestedAttribute to separate filter attribute and output attribute?

https://github.com/folio-sec/terraform-provider-zoom/blob/main/internal/services/phone/phonenumbers/phone_numbers_data_source.go#L60

```golang
  "filter": schema.SingleNestedAttribute {
    "status": ...
    "role_id": ...
  }
  "users: ...
```

Reply:
I have no objection to nesting.
On the other hand, the parameter `include_fields` is closer in meaning to **select** than to **filter**.
Therefore, the safe name `query` may be acceptable.

What do you think this?

Reply:
`include_fields`'s use-case is interesting. We should make it flexisible on data source. Let's migrate it using `query`.

Reply:
I have fixed the attribute. pls check it.